examples/page-object-model.md

```diff
@@ -0,0 +1,158 @@
+---
+description: A guide to implementing Page Object Model within Maestro
+---
+
+# Page Object Model
+
+
+## Content
+
+* Introduction
+* Login Page example
+  * login.js
+* Nested example
+  * cards.js
+* How to structure and tips
+  * Folder structure example
+  * Ensuring all elements are loaded
+    * loadElements.yaml
+    * flow.yaml
+* Cross-platform example
+
+
+## Introduction
+
+Implementating a Page Object Model has many benefits:
+* Improved readability, abstraction and grouping
+* Reducing unnecessary duplication
+* Improving test/flow maintenance
+
+It allows you to update an element in one place which will then cascade throughout all of your tests/flows. This can save a lot of time when elements are modified and also assist in easier debugging of issues.
+
+
+
+## Login Page example
+
+### login.js
+```
+// login.js
+output.login = {
+	email: 'email_text',
+	password: 'password_text',
+	loginBtn: 'loginButton',
+	registerBtn: 'registerButton'
+}
+```
+
+You can then use these variables to reference element IDs, text or even other test data:
+
+```
+- runScript: login.js
+- tapOn:
+	id: ${output.login.email}
+- inputText: "simon@maestro.com"
+- tapOn:
+	id: ${output.login.password}
+- inputText: ${PASSWORD}
+- tapOn:
+	id: ${output.login.loginBtn}
+```
+
+
+
+## Nested example
+
+### cards.js
+```
+// cards.js
+output.cards = {
+	cardName: 'card_title',
+	cardImage: 'card',
+
+	// All Cards Screen
+	allCards: {
+		createNewBtn: 'btn_create',
+		cardImage: 'card_image',
+		cardName: 'card_title'
+	},
+
+	// Virtual Cards
+	virtualCards: {
+		createNewBtn: 'btn_create',
+
+		cardDetails: {
+			cardNumber: 'card_number',
+			expiryDate: 'card_expiry'
+		}
+	}
+}
+```
+
+You can create nested element structures to better separate/organise screens and flows. Using the example below, it's easy to define a separate structure for Virtual Cards within the overall Cards object. You could then use them in your flows as follows:
+
+```
+- runScript: cards.js
+- tapOn:
+	id: ${output.cards.virtualCards.createNewBtn}
+- assertVisible:
+	id: ${output.cards.virtualCards.cardDetails.cardNumber}
+```
+
+
+
+## How to structure and tips
+
+### Folder structure example
+
+Creating an elements folder is good practice as it allows you to keep all of your element files together. 
+
+![maestro_element_structure](https://github.com/simon-gilmurray/maestro-docs/assets/74550141/5a95fb4c-41f4-42cb-83a6-d093e118a436)
```

Comment:
Nit: any chance you could update this to use a code block showcasing the folder structure instead of an image? That makes it easier if we ever need to change it in the future.